File: 0682-baseball-game/0682-baseball-game.java
class Solution {
    public int calPoints(String[] operations) {
        int result = 0;
        Deque<Integer> stack = new ArrayDeque<>();
        for (String op : operations) {
            if (op.equals("+")) {
                int last = stack.pop();
                int sum = last + stack.peek();
                stack.push(last);
                stack.push(sum);
                result += sum;
            } else if (op.equals("D")) {
                int score = stack.peek() * 2;
                stack.push(score);
                result += score;
            } else if (op.equals("C")) {
                result -= stack.pop();
            } else {
                int num = Integer.parseInt(op);
                stack.push(num);
                result += num;
            }
        }

        return result;
    }
}

File: 0682-baseball-game/0682-baseball-game.kt
class Solution {
    fun calPoints(operations: Array<String>): Int {
        var result = 0
        val stack = ArrayDeque<Int>()

        for (op in operations) {
            when (op) {
                "+" -> {
                    val last = stack.removeLast()
                    val sum = last + stack.last()
                    stack.addLast(last)
                    stack.addLast(sum)
                    result += sum
                }
                "D" -> {
                    val score = stack.last() * 2
                    stack.addLast(score)
                    result += score
                }
                "C" -> {
                    result -= stack.removeLast()
                }
                else -> {
                    val num = op.toInt()
                    stack.addLast(num)
                    result += num
                }
            }
        }

        return result
    }
}